CMakeLists.txt

```
@@ -69,6 +69,7 @@ openpmd_option(ADIOS1         "ADIOS1 backend (.bp files)"                AUTO)
 openpmd_option(ADIOS2         "ADIOS2 backend (.bp files)"                AUTO)
 openpmd_option(PYTHON         "Enable Python bindings"                    AUTO)
 
+option(openPMD_CXX_STANDARD          "Which C++ standard to use (14 or 17)" OFF)
```

Alternative suggestion:
```suggestion
openpmd_option(REGION         "Enable regions interfaces (C++17)"         AUTO)

```

And below we check:
```cmake
CMAKE_CXX_COMPILE_FEATURES
if(openPMD_USE_REGION STREQUAL AUTO)
    if ("cxx_std_17" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
        set(openPMD_HAVE_REGION ON)
    else()
        set(openPMD_HAVE_REGION OFF)
    endif()
elseif(openPMD_USE_REGION)
    set(openPMD_HAVE_REGION ON)
else()
    set(openPMD_HAVE_REGION OFF)
endif()
```

This way we:
- get the feature by default for all new compilers
- can enforce enable/disable of the feature so it does not accidentally disable itself, e.g. for package managers and CI
- keep full control & standardization over C++14/17/20/... via `CMAKE_CXX_STANDARD` options